### DISCUSSION TITLE
Could `TAILWIND_CLI_PATH` point to a binary?

### DISCUSSION THREAD

**Discussion by jefftriplett** [oliverandrich/django-tailwind-cli#65] [General]
Wonderful tool btw. 

This is a one-part question and one-part help for anyone else who might be trying to wrap their head around making this work in Docker so that it doesn't try to re-download the tailwindcss binary every time `python manage.py tailwind <cmd>` is run. 

I'm using `django-tailwind-cli` in containers, and I found myself struggling/jumping through a few hoops to get the `tailwindcss` binary to work effectively. 

I mistakenly had `TAILWIND_CLI_PATH` set to `/usr/bin/tailwindcss` which created `/usr/bin/tailwindcss/tailwindcss-linux-x64-3.3.5`.  

What I'm using now is:

```Dockerfile
ARG TAILWINDCSS_VERSION=3.3.5
...

# download the Tailwind CSS CLI binary
ADD https://github.com/tailwindlabs/tailwindcss/releases/download/v${TAILWINDCSS_VERSION}/tailwindcss-linux-x64 \
    /usr/bin/tailwindcss-linux-x64-${TAILWINDCSS_VERSION}

RUN chmod 755 /usr/bin/tailwindcss-linux-x64-${TAILWINDCSS_VERSION}

...

RUN DATABASE_URL=sqlite://:memory: python manage.py tailwind --skip-checks build
```

Would it make sense to allow users to set where the Tailwind CSS CLI binary is stored?

**Comment by oliverandrich** [comment]
The reasoning of using `TAILWIND_CLI_PATH` as the base path where to store the actual binary is, that if you are working with a team of developers using a mixed setup of operating systems and system architectues, as we do, we can share a common configuration for the project, and everyone gets the correct version of the CLI installed.

But I can see the issue you are facing right now. We are not using dev containers at all, so we haven't encountered it yet.

Two quick ideas:
- If `TAILWIND_CLI_PATH` is pointing at a directory, we stick to the current behaviour.
- If `TAILWIND_CLI_PATH` is referencing an existing file, the file is checked and used as the CLI of choice.

This would allow you to mount a common CLI version into your dev containers, or base them on a common base image which already includes the CLI version.

What do you think?

**Reply by jefftriplett** [reply]
I think `TAILWIND_CLI_PATH` being able to point to a binary would be useful. 

That said, the more I thought about it, the more I realized there wasn't a huge reason to manually download the binary like I was doing when `django-tailwind-cli` could do in my `Dockerfile`. 

For anyone reading up on this. This is good enough:

```Dockerfile
...

# add this towards the bottom
RUN DATABASE_URL=sqlite://:memory: python manage.py tailwind --skip-checks build

...
```

I think because I was already downloading the tailwind binary, I was approaching the problem differently. 

[[Fwiw, I'm not using `dev containers` (boo), but I'm using `docker compose` (yay) and the development flow is really nice for all platforms.]]

Thanks again!

**Reply by oliverandrich** [reply]
Great. And I will add the extra functionality for the `TAILWIND_CLI_PATH`.
